app/src/main/java/edu/wpi/first/shuffleboard/app/components/DashboardTabPane.java

```diff
@@ -211,15 +215,40 @@ private void populate() {
 
           // Don't create widgets for the catchall types
           if (source.getDataType() != DataTypes.Unknown
-              && source.getDataType() != DataTypes.Map
-              && !Components.getDefault().widgetNamesForSource(source).isEmpty()) {
-            Components.getDefault().createWidget(Components.getDefault().widgetNamesForSource(source).get(0), source)
-                .ifPresent(w -> getWidgetPane().addWidget(w));
+              && source.getDataType() != DataTypes.Map) {
+            Components.getDefault().defaultWidgetNameFor(source.getDataType())
+                .flatMap(n -> Components.getDefault().createComponent(n))
+                .map(c -> {
+                  if (c instanceof Widget) {
+                    ((Widget) c).setSource(source);
+                  } else {
+                    c.titleProperty().setValue(source.getName());
+                  }
+                  return c;
+                })
+                .filter(c -> noDuplicateTiles(c))
+                .ifPresent(component -> {
+                  getWidgetPane().components()
+                      .filter(c -> c instanceof ComponentContainer)
+                      .map(c -> (ComponentContainer & Component) c)
+                      .filter(c -> source.getName().startsWith(c.getTitle()) && !c.getTitle().equals(source.getName()))
+                      .findFirst()
+                      .map(c -> (ComponentContainer) c)
+                      .orElseGet(this::getWidgetPane)
+                      .addComponent(component);
+                });
```

Comment:
I thought you were going to refactor this into a different file?

Reply:
That didn't pan out.